form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt

```diff
@@ -776,6 +786,61 @@ private fun ProvideQRInput(
     )
 }
 
+@Composable
+private fun ProvideOrgUnitInput(
+    modifier: Modifier,
+    fieldUiModel: FieldUiModel,
+    uiEventHandler: (RecyclerViewUiEvents) -> Unit,
+    intentHandler: (FormIntent) -> Unit,
+) {
+    var inputFieldValue by remember(
+        fieldUiModel,
+    ) {
+        mutableStateOf(fieldUiModel.displayName)
+    }
+
+    InputOrgUnit(
+        modifier = modifier.fillMaxWidth(),
+        title = fieldUiModel.label,
+        state = fieldUiModel.inputState(),
+        supportingText = fieldUiModel.supportingText(),
+        legendData = fieldUiModel.legend(),
+        inputText = inputFieldValue ?: "",
+        isRequiredField = fieldUiModel.mandatory,
+        onValueChanged = {
+            inputFieldValue = it
+            intentHandler(
+                FormIntent.OnSave(
+                    fieldUiModel.uid,
+                    it,
+                    fieldUiModel.valueType,
+                ),
+            )
+        },
+        onOrgUnitActionCLicked = {
+            uiEventHandler.invoke(
+                RecyclerViewUiEvents.OpenOrgUnitDialog(
+                    fieldUiModel.uid,
+                    fieldUiModel.label,
+                    fieldUiModel.value,
+                ),
+            )
+        },
+        onFocusChanged = { focused ->
```

Comment:
do we need onFocusChange?

Reply:
Mm not hundred percent sure, I'll double check it and remove it if it is not necessary